src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -74,58 +77,75 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
 
   val userServiceRoutes =
     path("me") {
-      get { requestContext =>
+      parameter("userDetailsOnly".?) { userDetailsOnly =>
+        get { requestContext =>
 
-        // inspect headers for a pre-existing Authorization: header
-        val authorizationHeader: Option[HttpCredentials] = (requestContext.request.headers collect {
-          case Authorization(h) => h
-        }).headOption
+          // inspect headers for a pre-existing Authorization: header
+          val authorizationHeader: Option[HttpCredentials] = (requestContext.request.headers collect {
+            case Authorization(h) => h
+          }).headOption
 
-        authorizationHeader match {
-          // no Authorization header; the user must be unauthorized
-          case None =>
-            respondWithErrorReport(Unauthorized, "No authorization header in request.", requestContext)
-          // browser sent Authorization header; try to query Sam for user status
-          case Some(_) =>
-            val pipeline = authHeaders(requestContext) ~> sendReceive
-            val samRequest = Get(UserApiService.samRegisterUserURL)
-            pipeline(samRequest) onComplete {
-              case Success(response: HttpResponse) =>
-                response.status match {
-                  // Sam rejected our request. User is either invalid or their token timed out; this is truly unauthorized
-                  case Unauthorized =>
-                    respondWithErrorReport(Unauthorized, "Request rejected by identity service - invalid user or expired token.", requestContext)
-                  // Sam 404 means the user is not registered with FireCloud
-                  case NotFound =>
-                    respondWithErrorReport(NotFound, "FireCloud user registration not found.", requestContext)
-                  // Sam error? boo. All we can do is respond with an error.
-                  case InternalServerError =>
-                    respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again.", requestContext)
-                  // Sam found the user; we'll try to parse the response and inspect it
-                  case OK =>
-                    val respJson = response.entity.as[RegistrationInfo]
-                    respJson match {
-                      case Right(regInfo) =>
-                        if (regInfo.enabled.google && regInfo.enabled.ldap && regInfo.enabled.allUsersGroup) {
-                          // Sam says the user is fully registered and activated!
-                          requestContext.complete(OK, regInfo)
-                        } else {
-                          // Sam knows about the user, but the user isn't activated
-                          respondWithErrorReport(Forbidden, "FireCloud user not activated.", requestContext)
+          authorizationHeader match {
```

Comment:
I'm trying to figure out what you changed vs. what git _thinks_ you changed.  I think I count at least 4 neseted case statements, on top of the normal akka nesting.  With caution, I would recommend that if you're making changes here, try to start extracting these nested statements into their own methods.  

This is the sort of thing I'd normally reject a PR for, but seeing as it doesn't seem like you are leaving the code in any worse shape than when you started (I think, it's hard to tell, and that's part of the problem) I'm not going to worry too much about it.